src/applications/travel-pay/components/complex-claims/pages/ExpensePage.jsx

```diff
@@ -0,0 +1,198 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom-v5-compat';
+import {
+  VaModal,
+  VaButton,
+  VaButtonPair,
+  VaDate,
+  VaTextInput,
+} from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+import { EXPENSE_TYPES } from '../../../constants';
+import ExpenseMealFields from './ExpenseMealFields';
+import ExpenseAirTravelFields from './ExpenseAirTravelFields';
+import ExpenseLodgingFields from './ExpenseLodgingFields';
+import ExpenseCommonCarrierFields from './ExpenseCommonCarrierFields';
+
+const ExpensePage = () => {
+  const navigate = useNavigate();
+  const { apptId, claimId, expenseTypeRoute } = useParams();
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [formState, setFormState] = useState({});
+  const [showError, setShowError] = useState(false);
+  const errorRef = useRef(null); // ref for the error message
+
+  // Focus the error message when it becomes visible
+  useEffect(
+    () => {
+      if (showError && errorRef.current) {
+        errorRef.current.focus();
+      }
+    },
+    [showError],
+  );
+
+  const expenseType = Object.keys(EXPENSE_TYPES).find(
+    key => EXPENSE_TYPES[key].route === expenseTypeRoute,
+  );
+
+  const expenseTypeFields = expenseType ? EXPENSE_TYPES[expenseType] : null;
+
+  const handleFormChange = (event, explicitName) => {
+    // Figure out the field name
+    const name = explicitName ?? event.target?.name ?? event.detail?.name; // rarely used, but safe to include
+
+    // Figure out the value (covers VA components + normal inputs)
+    const value =
+      event?.value ?? event?.detail?.value ?? event.target?.value ?? '';
+
+    setFormState(prev => ({
+      ...prev,
+      [name]: value,
+    }));
+  };
+
+  const handleOpenModal = () => setIsModalVisible(true);
+  const handleCloseModal = () => {
+    setIsModalVisible(false);
+    navigate(`/file-new-claim/${apptId}/${claimId}/review`);
+  };
+
+  const REQUIRED_FIELDS = {
+    Meal: ['vendor'],
+    Lodging: ['vendor', 'checkInDate', 'checkOutDate'],
+    Commoncarrier: ['transportationType', 'transportationReason'],
+    Airtravel: [
+      'vendorName',
+      'tripType',
+      'departureDate',
+      'departureAirport',
+      'arrivalDate',
+      'arrivalAirport',
+    ],
+  };
+
+  const validatePage = () => {
+    const base = ['date', 'amount'];
+    const extra = REQUIRED_FIELDS[expenseType] || [];
+    const requiredFields = [...base, ...extra];
+
+    const emptyFields = requiredFields.filter(field => !formState[field]);
+
+    setShowError(emptyFields.length > 0);
+    return emptyFields.length === 0;
+  };
+
+  const handleContinue = () => {
+    const isValid = validatePage();
+    if (!isValid) return; // stop navigation if validation fails
+    navigate(`/file-new-claim/${apptId}/${claimId}/review`);
```

Comment:
Re: our conversation about the review alert, this is what push state might look like to provide a flag on the review page for the alert that the expense was added successfully or not.
```js
    navigate(`/file-new-claim/${apptId}/${claimId}/review`, {
      state: { success: [true|false] }
    });
```
```js
const location = useLocation();
const isSuccessful = location.state.success;
```